### DISCUSSION TITLE
Split into different modules

### DISCUSSION THREAD

**Discussion by RuinTD** [Aktanusa/CookieMonster#958] [Ideas]
Since the CM Team is delving into brand new mods, I was thinking it might be a good idea to split CM into separate modules. Most notable thing to come to mind could be splitting the notifications feature into a mod called CM Notify. Anyone who uses CM for the features other than notifications could just choose not to install CM Notify instead of having to go in and disable all the settings. However, anyone who wants the notifications could install CM Notify and everything will be enabled by default, which would be ideal.

In addition, this would make it easier for Steam users to use Cookie Monster and only have modules installed that **don’t** prevent achievements. Anything in CM that could be considered “cheating” could be in a separate module that would explicitly disable achievements if loaded on the Steam version.

If there’s any concern for messing with users’ experience having to download mods, we could allow the main CM mod to automatically load CM modules and basically act as a mod manager for CM (or potentially for third-party mods). For any existing users of CM when this theoretical update pushes, CM could automatically have all the modules enabled and each modules could run a migration script on first run to move all of its settings out of CM’s settings. Then the user could choose to disable modules as they desire and remove their individual mod data. This CM mod manager could even include CM Shortcuts or any other future mods that CM Team comes out with.

**Comment by DanielNoord** [comment]
Splitting the CM Notify mod would indeed be something I would want to look into. We could use the Cookie Clicker Mod Manager to load different modules on the browser.
However, I would want to investigate how easy this is to do for mods for the Steam workshop and if there are any potential issues there.
